Switch to non-blocking thread for handling search task

RussB-d3t opened the pull request:
- Rework existing external process handling to avoid blocking the UI thread during search.
- Add cancellation token for handing cancellation of search task including clean up of external process.
- Add ViewModel and bindings to the tool window, update window XAML to support search/cancel and a progress bar for user feedback.
- Refactor to remove compilation/analyzer warnings regarding threading.

RussB-d3t committed:
Switch to non-blocking thread for handling search task

- Rework existing external process handling to avoid blocking the UI thread during search.
- Add cancellation token for handing cancellation of search task including clean up of external process.
- Add ViewModel and bindings to the tool window, update window XAML to support search/cancel and a progress bar for user feedback.
- Refactor to remove compilation/analyzer warnings regarding threading.

Josef-Rare commented on Source/BlueprintSearchVSExtension/Source/Commands/CommandHandlers/ExecuteSearchHandler.cs:
```diff
@@ -6,24 +6,32 @@
 		private const string CommandLineArguments = "-NoShaderCompile -SILENT -run=FiB";
 
-		public List<BlueprintJsonObject> MakeSearch(string InSearchValue)
+		public async Task MakeSearchAsync(string InSearchValue, CancellationToken cancellationToken)
```
Well, the coding standards are a bit all over the place in this project I can see. But can this cancellationToken have a capital C please so at least this file is consistent.

Josef-Rare commented on Source/BlueprintSearchVSExtension/Source/Commands/CommandHelpers/PathFinderHelper.cs:
```diff
@@ -27,82 +27,127 @@ public static class PathFinderHelper
+			Microsoft.VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread();
+
 			List<Project> ProjectsList = new List<Project>();
 			ProjectsList.AddRange(DTEService.Solution.Projects.Cast<Project>());
```
DTE compat is one of the things that might prove a headache when moving the plugin to 2022 (no action on this review just something you might want to be aware of)

Josef-Rare commented on Source/BlueprintSearchVSExtension/Source/UI/ToolWindows/BlueprintSearchWindow/BlueprintSearchVSWindowControl.xaml.cs:
```diff
@@ -45,13 +44,21 @@ private void OnKeyDownHandler(object InSenderObject, KeyEventArgs InEventArgs)
+		/// Handles click on the button by displaying a message box.
+		/// </summary>
+		/// <param name="sender">The event sender.</param>
+		/// <param name="e">The event args.</param>
```
These names should reflect the func param names (right?)

Josef-Rare commented on Source/BlueprintSearchVSExtension/Source/UI/ToolWindows/BlueprintSearchWindow/BlueprintSearchVSWindowVM.cs:
```diff
@@ -0,0 +1,152 @@
+using BlueprintSearch.Commands.CommandHandlers;
```
// Copyright (C) Coconut Lizard Limited. All rights reserved.
// ---------------------------------------------------------

To top of file

Josef-Rare commented on Source/BlueprintSearchVSExtension/Source/UI/ToolWindows/BlueprintSearchWindow/BlueprintSearchVSWindowVM.cs:
```diff
@@ -0,0 +1,152 @@
+			{
+				if (PathFinderHelper.FindUEProject(out string UEProjectPath))
+				{
+					if (PathFinderHelper.FindPaths(UEProjectPath) == false)
```
could just `return PathFinderHelper.FindPaths(UEProjectPath);` here

Josef-Rare reviewed (changes requested):
Couple of trivial comments otherwise looks good!

RussB-d3t committed:
Fixing issues raised by review

- updated incorrectly named parameters
- added copyright notice to added file
- updated summary comments
- simplified function return cases
